common/djangoapps/util/course.py

```diff
@@ -0,0 +1,31 @@
+"""
+Utility methods related to course
+"""
+import logging
+from django.conf import settings
+
+from opaque_keys.edx.keys import CourseKey
+import re
+
+log = logging.getLogger(__name__)
+
+
+def get_lms_link_for_about_page(course_key):
+    """
+    Returns the url to the course about page from the location tuple.
+    """
+    assert isinstance(course_key, CourseKey)
+
+    if settings.FEATURES.get('ENABLE_MKTG_SITE', False):
+        # Root will be "https://www.edx.org". The complete URL will still not be exactly correct,
+        # but redirects exist from www.edx.org to get to the Drupal course about page URL.
+        about_base = settings.MKTG_URLS.get('ROOT')
+    else:
+        about_base = settings.ENV_TOKENS.get('LMS_BASE')
+
+    # Strip off https:// (or http://) to be consistent with the formatting of LMS_BASE.
+    about_base = re.sub(r"^https?://", "", about_base)
+    return u"https://{about_base_url}/courses/{course_key}/about".format(
```

Comment:
I still hate this.  We're making assumptions about the URL scheme that are unvalidated.  `https` is a safer assumption than `http` from the perspective of user data, but it can still lead to incorrect behavior, depending on how sites are configured.

I spoke with devops about this, and they recommended using a new setting, `LMS_ROOT_URL` that contains the protocol information.  I'll add a commit to this PR to add that setting, then if you use that instead of `LMS_BASE`, there won't be a need to strip URL schemes, and guess what value to use.